modules/standard/Buffers.chpl

```diff
@@ -140,18 +141,34 @@ module Buffers {
      :arg error: (optional) capture an error that was encountered instead of
                  halting on error
    */
-  proc bytes.bytes(len:int(64), out error:syserr) {
+  proc bytes.init(len:int(64), out error:syserr) {
+    this.home = here;
+    super.init();
     error = qbytes_create_calloc(this._bytes_internal, len);
     // The buffer is "retained" internally on creation, but only on success.
-    this.home = here;
   }
   pragma "no doc"
-  proc bytes.bytes(len:int(64)) {
-    var error:syserr = ENOERR;
-    error = qbytes_create_calloc(this._bytes_internal, len);
+  proc bytes.init(len:int(64)) {
+    this.home = here;
+    super.init();
+    var error:syserr = qbytes_create_calloc(this._bytes_internal, len);
     if error then try! ioerror(error, "in bytes constructor");
     // The buffer is retained internally on construction, but only on success.
+  }
+
+
+  pragma "no doc"
+  proc bytes.init(x: bytes) {
     this.home = here;
+    if x.home == here {
+      this._bytes_internal = x._bytes_internal;
+      super.init();
+      qbytes_retain(x._bytes_internal);
```

Comment:
Note that this could also say `qbytes_retain(this._bytes_internal)`. I think that'd be clearer, but it shouldn't matter.

Reply:
Actually on second thought the ideal story for reference counting is to increase the reference count immediately once we are in the function. Can we do `qbytes_retain(x._bytes_internal);` before setting `this._bytes_internal` ?
But all of this feedback is probably in the noise - you could go ahead as-is.